libavformat/dashenc.c

```diff
@@ -2415,8 +2415,11 @@ static int dash_write_packet(AVFormatContext *s, AVPacket *pkt)
 
     int64_t frame_duration_variation = 0;
 
-    if (s->prev_pts != AV_NOPTS_VALUE)
+    if (s->prev_pts != AV_NOPTS_VALUE) {
         frame_duration_variation = (s->max_frame_duration - s->min_frame_duration) * 2;
+        if (frame_duration_variation >= pkt->duration)
+            frame_duration_variation = pkt->duration - 1;
+    }
```

Comment:
In the case packet durations are 1501, 1502, this risks to be equal to the shorter frame (1501).

I think a safer option is `(pkt->duration / 2 + 1)`
Or `(pkg->duration - 2)`

Reply:
The key is not going beyond packet duration. The half + one also works.